### PR TITLE
Added Free O'Reilly Programming Books

### PR DESCRIPTION
**Pull request by lougeniaC64**
Added Free O'Reilly Programming books referred to in issue #2113.  

FYI: there are additional O'Reilly books in the web development, IoT, Data, Security, and WebOps pages.

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -636,19 +644,28 @@
 * [Architectural Styles and the Design of Network-based Software Architectures](http://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm) - Roy Thomas Fielding
 * [Best Kept Secrets of Peer Code Review](http://smartbear.com/lp/ebook/collaborator/secrets-of-peer-code-review/)
 * [Data-Oriented Design](http://www.dataorienteddesign.com/dodmain/dodmain.html)
+* [Developing Reactive Microservices](https://info.lightbend.com/COLL-20XX-Developing-Reactive-Microservices_Landing-Page.html?lst=OR)
 * [Domain Driven Design Quickly](http://www.infoq.com/minibooks/domain-driven-design-quickly)
 * [DSL Engineering: Designing, Implementing and Using Domain-Specific Languages](http://dslbook.org) - Markus Voelter
+* [Essential Skills for Agile Development](http://elliottback.com/wp/essential-skills-for-agile-development/)
```
Comment (eshellman) [inline free-programming-books.md]: website not responding

**Comment by lougeniaC64** [comment]
@eshellman Removed dead link and added annotations as requested.

**Comment by eshellman** [comment]
Thanks!

**Comment by lougeniaC64** [comment]
NP, glad to assist!

**Pull request merged**